packages/product/integration-tests/__tests__/services/product/index.ts

```diff
@@ -412,6 +421,129 @@ describe("Product Service", () => {
       })
     })
 
+    describe("relation: collections", () => {
+      let workingProduct: Product
+      let workingProductTwo: Product
+      let workingCollection: ProductCollection
+      let workingCollectionTwo: ProductCollection
+      const collectionData = [
+        {
+          id: "test-1",
+          title: "col 1",
+        },
+        {
+          id: "test-2",
+          title: "col 2",
+        },
+      ]
+
+      beforeEach(async () => {
+        testManager = await TestDatabase.forkManager()
+        collections = await createCollections(testManager, collectionData)
+        workingCollection = (await testManager.findOne(
+          ProductCollection,
+          "test-1"
+        )) as ProductCollection
+        workingCollectionTwo = (await testManager.findOne(
+          ProductCollection,
+          "test-2"
+        )) as ProductCollection
+
+        products = await createProductAndTags(testManager, [
+          {
+            ...productsData[0],
+            collection_id: workingCollection.id,
+          },
+          {
+            ...productsData[1],
+            collection_id: workingCollectionTwo.id,
+          },
+        ])
+
+        workingProduct = products.find((p) => p.id === "test-1") as Product
+        workingProductTwo = products.find((p) => p.id === "test-2") as Product
+      })
+
+      it("should filter by collection relation and scope fields", async () => {
+        const products = await service.list(
+          {
+            id: workingProduct.id,
+            collection_id: workingCollection.id,
+          },
+          {
+            select: ["title", "collection.title"],
+            relations: ["collection"],
+          }
+        )
+
+        const serialized = JSON.parse(JSON.stringify(products))
+
+        expect(serialized.length).toEqual(1)
+        expect(serialized).toEqual([
+          {
+            id: workingProduct.id,
+            title: workingProduct.title,
+            collection_id: workingCollection.id,
+            collection: {
+              id: workingCollection.id,
+              title: workingCollection.title,
+            },
+          },
+        ])
+      })
+
+      it("should filter by collection when multiple collection ids are passed", async () => {
```

Comment:
question: Could this be a false positive since we only have two collections and both products are part of them?

Reply:
good point, added one more product in there.